app/javascript/src/NewApplication/data/Products.js

```diff
@@ -0,0 +1,27 @@
+// @flow
+
+import { getJSON } from 'utilities'
+
+import type { Product, FetchItemsRequestParams, FetchItemsResponse } from 'NewApplication/types'
+
+const fetchPaginatedProducts = (path: string, params: FetchItemsRequestParams): FetchItemsResponse<Product> => {
+  const { page, perPage, query = '' } = params
+
+  const searchParams = new URLSearchParams({
+    page: String(page),
+    per_page: String(perPage),
+    sort: 'updated_at',
+    direction: 'desc'
+  })
+
+  if (query !== '') {
+    searchParams.append('search[query]', query)
+    searchParams.append('utf8', '✓')
+  }
+
+  const url = `${path}?${searchParams.toString()}`
+
+  return getJSON(url).then(data => data.json())
```

Comment:
Maybe some error handling here or in the method definition

Reply:
There was at first but then I thought it wasn't useful. If errors are handled at this level, then something like `undefined` or `[]` has to be returned and then the component won't know there's been an error.
By handling it at the component level we can update the UI with a meaningful error message, retry the request, etc.